### PR TITLE
Improve LaTeX literate backend

### PR DESCRIPTION
**Pull request by imkiva**
/cc @ice1000

### CODE REVIEW

File: cli-impl/src/main/java/org/aya/cli/literate/FaithfulPrettier.java [hunk 1]
```diff
@@ -59,19 +49,25 @@ private static void checkHighlights(@NotNull SeqLike<HighlightInfo> highlights)
...
+      var knifeCut = twoKnifeThreeParts(raw, base, current.sourcePos());
+      // move forward
+      raw = knifeCut.remaining;
+      base = knifeCut.base;
+
+      // If there's orphan text before the highlighted cut, add it to result as plain text.
+      if (!knifeCut.before.isEmpty()) {
+        // TODO: handle whitespaces in the lexer, and use a new highlight type for them.
+        //  this workaround solution does not work for whitespace in LaTeX.
+        var lines = knifeCut.before.toString().split("\n", -1);
```
Comment (imkiva) [inline cli-impl/src/main/java/org/aya/cli/literate/FaithfulPrettier.java:63]: ij-parsing-core seems not providing detailed whitespace kind, e.g. tabs, spaces, line breaks, etc.

Comment (ice1000) [reply cli-impl/src/main/java/org/aya/cli/literate/FaithfulPrettier.java:63]: Whitespaces can be ~ in LaTeX, and we are not using tabs for now

Comment (ice1000) [reply cli-impl/src/main/java/org/aya/cli/literate/FaithfulPrettier.java:63]: I think it's related to how our lexer is written. We are probably not distinguishing among various whitespace tokens

**Review by ice1000 — APPROVED** [review]
I'm convinced

**Comment by ice1000** [comment]
bors merge

File: pretty/src/main/java/org/aya/pretty/backend/latex/DocTeXPrinter.java [hunk 2]
```diff
@@ -40,13 +41,31 @@ public class DocTeXPrinter extends StringPrinter<DocTeXPrinter.Config> {
...
+    id("(|"), id("|)"),
+    Tuple.of("\u2987", "(|"),
+    Tuple.of("\u2988", "|)"),
     id("[|"), id("|]"),
+    Tuple.of("\u27E6", "[|"),
+    Tuple.of("\u27E7", "|]"),
+    Tuple.of("{|", "\\{|"),
+    Tuple.of("|}", "|\\}"),
+    Tuple.of("\u2983", "\\{|"),
+    Tuple.of("\u2984", "|\\}"),
```
Comment (ice1000) [inline pretty/src/main/java/org/aya/pretty/backend/latex/DocTeXPrinter.java:66]: I think we should use dedicated LaTeX symbols. For instance I guess we have `\lBrace` `\rBrace` something like dat

**Pull request merged**